Neos.Fusion/Classes/FusionObjects/DataStructureImplementation.php

```diff
@@ -38,7 +39,8 @@ public function evaluate()
         $result = [];
         foreach ($sortedChildFusionKeys as $key) {
             $propertyPath = $key;
-            if ($this->isUntypedProperty($this->properties[$key])) {
+            if ($this->isImplementationDataStructureImplementation()
+                && $this->isUntypedProperty($this->properties[$key])) {
                 $propertyPath .= '<Neos.Fusion:DataStructure>';
             }
```

Comment:
i agree on the `'<' . $this->fusionObjectName . ' >';` part  (see https://github.com/neos/neos-development-collection/pull/3568#issuecomment-1018747823)

the `$this->isImplementationDataStructureImplementation()` barrier is to limit the untyped feature only to datastructures - otherwise with all this inheritance stuff we might never know what we allow with this unwillingly: see  https://github.com/neos/neos-development-collection/issues/3576

the ugly `isImplementationDataStructureImplementation` can be removed once we refactor #3577

Reply:
How about having a defaultType configured in the data stucture implementation. 
That could be altered in derived implementations easyly 

```
class DataStructureImplementation extends AbstractArrayFusionObject
    protected $defaultFusionObjectType = 'Neos.Fusion:DataStructure';

    # used like:
    
    if ($this->isUntypedProperty($this->properties[$key]) &&  $this->defaultFusionObjectType) {
        $propertyPath .= '<' . $this->defaultFusionObjectType . ' >';
    }
```

```
class JoinImplementation extends DataStructureImplementation {
    protected $defaultFusionObjectType = 'Neos.Fusion:Join';
```

```
class HttpResponseImplementation extends DataStructureImplementation {
    protected $defaultFusionObjectType = null;
```

Reply:
Sounds like a nice idea.